src/main/webapp/app/overview/course-competencies/course-competencies-details.component.ts

```diff
@@ -37,13 +39,17 @@ export class CourseCompetenciesDetailsComponent implements OnInit {
     ) {}
 
     ngOnInit(): void {
-        this.activatedRoute.params.subscribe((params) => {
-            this.competencyId = +params['competencyId'];
-            this.courseId = +params['courseId'];
```

Comment:
What is the issue with the old approach? 
For me, it was easier to read and seems more robust (e.g. when changing the route) - what am I missing here?

Reply:
It's because of the new routing. Before we had some paths like` `'courses/:courseId/exercises/:exerciseId'`,`.This was less modular, and does not clearly show the relationship between the course and exercises. The new routing clearly represents the parent-child relationship between courses, exercises, and exercise details, which makes the route structure more intuitive. It's more modular and scalable. The only tiny downside is that you need to get the courseId from the parent-params.